### ISSUE TITLE
[Bug]: aws_api_gateway_method_settings logging_level = "ERROR"/"INFO" doen't work and data_trace_enabled is always set to ture

### ISSUE THREAD

**Issue by Mafuj-Jia** [hashicorp/terraform-provider-aws#27834]
### Terraform Core Version

Terraform v1.0.0

### AWS Provider Version

4.39.0

### Affected Resource(s)

**aws_api_gateway_method_settings**
  settings {
    metrics_enabled = true
    logging_level   = "ERROR"
    data_trace_enabled = false
  }

### Expected Behavior

api_gateway stage cloudwatch loglevel "ERROR" should change according to 
**aws_api_gateway_method_settings**
  settings {
    metrics_enabled = true
    logging_level   = "ERROR"
    data_trace_enabled = false
  }

### Actual Behavior

But whatever the log_level could be selected except "OFF" in terraform above code snippet, it's always set to "Full Request And Response Log" (verified from apigateway > stage > Log Tracing menu > cloudwatch log_level)

### Relevant Error/Panic Output Snippet

```shell
There is no error output printed.
```

### Terraform Configuration Files

resource "aws_api_gateway_method_settings" "default" {
  count = var.deployment_enabled ? 1 : 0

  rest_api_id = aws_api_gateway_rest_api.default.*.id[0]
  stage_name  = var.stage_name
  method_path = "*/*"

  settings {
    metrics_enabled    = true
    data_trace_enabled = false
    logging_level      = "ERROR"
  }

  depends_on = [
    aws_api_gateway_rest_api.default,
    aws_api_gateway_stage.default,
  ]
}

### Steps to Reproduce

terraform init and terraform apply

### Debug Output

Test 2022-11-16T09:42:22Z logger.go:66:   ~ resource "aws_api_gateway_method_settings" "default" {
Test 2022-11-16T09:42:22Z logger.go:66:         id          = "xxxxxx-dev-*/*"
Test 2022-11-16T09:42:22Z logger.go:66:         # (3 unchanged attributes hidden)
Test 2022-11-16T09:42:22Z logger.go:66: 
Test 2022-11-16T09:42:22Z logger.go:66:       ~ settings {
Test 2022-11-16T09:42:22Z logger.go:66:           ~ data_trace_enabled                         = true -> false
Test 2022-11-16T09:42:22Z logger.go:66:           ~ logging_level                              = "OFF" -> "ERROR"
Test 2022-11-16T09:42:22Z logger.go:66:             # (8 unchanged attributes hidden)
Test 2022-11-16T09:42:22Z logger.go:66:         }
Test 2022-11-16T09:42:22Z logger.go:66:     }

### Important Factoids

My suspect is **data_trace_enabled  = true/false**  becomes always true. It should be false according to parameter.

**Comment by Trentondmiller** [comment]
Can confirm, this is also happening for me:

Terraform version 1.3.6
AWS Provider 4.26.0

**Comment by amarjandu** [comment]
This happened to us too, caused data to leak into the logs :(

**Comment by amarjandu** [comment]
@Trentondmiller do you have `*/*` aws_api_gateway_method_settings block in your code?

**Comment by sap147** [comment]
Is this still resolved? We are seeing the same behavior for our APIGW terraform as well.

To clarify:

1. If you set `logging_level` to `ERRORS` `data_trace_enabled` to `true`, this will result in constant drift in your TF plans.

`logging_level` set to `ERROR` results in the cloudwatch_logs setting for your APIGW stage to be set to "Error Only" , this is  expected and what we want.

Now, if you have `data_trace_enabled` set to `true` in your terraform, and then you re-run the plan, you will see a drift indicating something such as below:

```
      ~ settings {
          ~ data_trace_enabled                         = false -> true
```
 
 If you apply the above change, it will impact the cloudwatch `logging_level` and change your cloudwatch logging level to `Full request and response logs` .. so its like two different settings are touching the same thing kind of (`logging_level` and `data_traced_enabled`).

**Comment by amarjandu** [comment]
@sap147 This was an issue for us, the problematic setup was when we had one resource "aws_api_gateway_method_settings" with `method_path= "*/*"` and then specifying overrides to this on endpoints that we require to be more locked down.

Rather than this our new approach was to just programmatically create a `aws_api_gateway_method_settings` for each endpoint, and remove the `"*/*"` resource, I have not seen the drift occur anymore.

**Comment by Mafuj-Jia** [comment]
The issue has been resolved by upgrading TF version to 1.5+.
